app/experimenter/docs/openapi-schema.json

```diff
@@ -7639,6 +7639,10 @@
                         "AUDIENCE 1",
                         "AUDIENCE 2"
                       ]
+                    },
+                    "bugzilla_url": {
+                      "type": "string",
+                      "readOnly": true
```

Comment:
Probably irrelevant but I found this elsewhere in the schema, and it looked interesting/relevant. Not sure if the `bugzilla_url` should also have a `"format": "uri"` or `pattern` defined.

```js
                      "feature_bugzilla_url": {
                        "type": "string",
                        "format": "uri",
                        "nullable": true,
                        "maxLength": 200,
                        "pattern": "^(?:[a-z0-9\\.\\-\\+]*)://(?:[^\\s:@/]+(?::[^\\s:@/]*)?@)?(?:(?:25[0-5]|2[0-4]\\d|[0-1]?\\d?\\d)(?:\\.(?:25[0-5]|2[0-4]\\d|[0-1]?\\d?\\d)){3}|\\[[0-9a-f:\\.]+\\]|([a-z\u00a1-\uffff0-9](?:[a-z\u00a1-\uffff0-9-]{0,61}[a-z\u00a1-\uffff0-9])?(?:\\.(?!-)[a-z\u00a1-\uffff0-9-]{1,63}(?<!-))*\\.(?!-)(?:[a-z\u00a1-\uffff-]{2,63}|xn--[a-z0-9]{1,59})(?<!-)\\.?|localhost))(?::\\d{2,5})?(?:[/?#][^\\s]*)?\\Z"
                      },
```

Reply:
@pdehaan `openapi-schema.json` is automatically generated so in general I'd recommend treating it more like a build artifact than source code.

Reply:
Good point, although now I guess I'm more confused as to why it wouldn't be more consistenter. But don't take my general curiosity and ignorance as a merge blocker.

Reply:
For the sake of curiosity, the field on teh serializer you're looking at is defined here:

https://github.com/mozilla/experimenter/blob/d654aba1d65a36772fd27387375b922f9af19880/app/experimenter/experiments/api/v1/serializers.py#L174

which is just referencing the model field without explicitly declaring any serializer field interface, so DRF will just try to automatically build a serializer field out of whatever the modelfield says, which is:

https://github.com/mozilla/experimenter/blob/151f9612d802414c736a47f0c639e19a071ba55d/app/experimenter/experiments/models.py#L238

a Django URLField, and so that will include the regex validation.

But there's a weird nuance in the way we treat different bugzilla tickets, so for the feature bugzilla ticket, we only care about the URL since we just show it as a link.   But for the experiment bugzilla ticket, we need to store the actual ID:

https://github.com/mozilla/experimenter/blob/151f9612d802414c736a47f0c639e19a071ba55d/app/experimenter/experiments/models.py#L232

since we use that to make API calls to bugzilla to manipulate the ticket, and then we reconstruct the URL manually:

https://github.com/mozilla/experimenter/blob/151f9612d802414c736a47f0c639e19a071ba55d/app/experimenter/experiments/models.py#L337-L340

so we never store the experiment bugzilla ticket as an actual url, which is why, when we serialize it, Django/DRF don't know to treat it as a url, and so there is no fancy regex ;)

Reply:
Also the regex would be used to validate incoming values, but we're treating it as a ready only field here, so you don't need the regex anyways ;)